gateway-spi/src/main/java/org/apache/knox/gateway/dispatch/DefaultHttpClientFactory.java

```diff
@@ -369,4 +374,12 @@ private static long parseTimeout( String s ) {
     Period p = Period.parse( s, f );
     return p.toStandardDuration().getMillis();
   }
+
+  private static String getCookieSpec(FilterConfig filterConfig) {
+    GatewayConfig globalConfig =
+            (GatewayConfig)filterConfig.getServletContext().getAttribute(GatewayConfig.GATEWAY_CONFIG_ATTRIBUTE);
+    return globalConfig != null
```

Comment:
This will always return use the gateway-level value (very likely `null`) even if `httpclient.cookieSpec` is set on the topology level. Based on the other helper methods here (`getMaxConnections`, `getConnectionTimeout`, `getSocketTimeout`,...) you should change this to check if `httpclient.cookieSpec` is set in the topology and default to the GatewayConfig value otherwise.

You may also want to refactor that code out to some private method to avoid boilerplate private methods.